integration_tests/models/behavior/verify_if_masking_policies_applied_correctly.sql

```diff
@@ -8,22 +8,40 @@
 {% if execute %}
     {%- set dbt_project_tags = dbt_tags.get_dbt_tags() %}
 {% endif %}
+{% set policy_data_types_list = var('dbt_tags__policy_data_types', []) -%}
 {%- set ns = dbt_tags.get_resource_ns() %}
 {%- set database_name = ns.split('.')[0] %}
 {%- set schema_name = ns.split('.')[1] %}
 
 with dbt_project_tags as (
 
-    {%- for item in dbt_project_tags if ".column" in item['level'] -%}
+    {%- for item in dbt_project_tags if (".column" in item['level'] and dbt_tags.get_masking_policy_for_tag(item.tag)) -%}
 
     {%- set masking_policy = dbt_tags.get_masking_policy_for_tag(item.tag) %}
       {%- if masking_policy %}
-        {%- set masking_policy_name = masking_policy.get_name().split('__')[1] %}
+        {% for policy_data_types in policy_data_types_list if item.tag in policy_data_types.keys() %}
```

Comment:
I'm seeing duplicates in the table. Perhaps there should be a distinct select afterwards?
<img width="1183" alt="image" src="https://github.com/user-attachments/assets/c4143913-db5d-4fe8-ad35-23e463ce4f9f">

Reply:
As they were only test models and there were only a few rows I didn't bother fixing this as there was no performance benefit or test result change. But I have added a distinct to the test models now to avoid these duplicates.